spec/integration/user_index_spec.rb

```diff
@@ -0,0 +1,64 @@
+require 'rails_helper'
+
+RSpec.describe 'Test the Main Index page', type: :feature do
+  describe 'GET the index page' do
+    before(:each) do
+      @first_user = User.create(
+        name: 'Victor',
+        photo: 'https://ibedc.com/filesupload/imagebuild/625350ecda1e22.55685312.jpg',
+        bio: 'Teacher',
+        post_counter: 14
+      )
+      @first_user.save!
+      @another_user = User.create(
+        name: 'John',
+        photo: 'https://ibedc.com/filesupload/imagebuild/625350c770c886.68494374.jpg',
+        bio: 'Developer',
+        post_counter: 2
+      )
+      @another_user.save!
+
+      visit root_path
+
+      @first_user_post = Post.create(
+        title: 'My First Post',
+        text: 'Victor First Post',
+        comments_count: 0,
+        likes_count: 3,
+        author: @first_user
+      )
+
+      @second_user_post = Post.create(
+        title: 'My Second Post',
+        text: 'John Second Post',
+        comments_count: 0, likes_count: 0,
+        author: @another_user
+      )
+
+      @first_comment = Comment.create(text: 'Victor\'s Comment', author: User.first, post: @first_user_post)
+      @second_comment = Comment.create(text: 'Johns Comments', author: User.first, post: @first_user_post)
+    end
+
+    # it "I can see the profile picture for each user." do
```

Comment:
Please remove commented code from here and also check other files as it is against the rule of clean code.